include/envoy/http/async_client.h

```diff
@@ -38,7 +38,40 @@ class AsyncClient {
   };
 
   /**
-   * An in-flight HTTP request
+   * Notifies caller of async HTTP stream status.
```

Comment:
It might be worth having some exposition about the full-duplex nature of streams in the documentation. For instance StreamCallbacks can continue to receive events even if the Stream has had sendXXX(..., end_stream=true) called.